Tests/scripts/validate_premium_packs.py

```diff
@@ -0,0 +1,210 @@
+"""Validate premium packs.
+
+Check index.json file inside the index.zip archive in the cloud.
+Validate no missing ids are found and that all packs have a positive price.
+Validate commit hash is in master's history.
+Check the server configured on master.
+Validate the pack id's in the index file are present on the server and the prices match.
+"""
+import demisto_client
+import argparse
+import logging
+import zipfile
+import json
+import ast
+import sys
+import os
+
+from Tests.configure_and_test_integration_instances import Build, Server
+from Tests.scripts.utils.log_util import install_logging
+from Tests.test_content import get_json_file
+
+INDEX_FILE_PATH = 'index.json'
+
+
+def options_handler():
+    parser = argparse.ArgumentParser(description='Utility for instantiating integration instances')
+    parser.add_argument('--ami_env', help='The AMI environment for the current run. Options are '
+                                          '"Demisto 6.0", "Demisto Marketplace". The server url is determined by the'
+                                          ' AMI environment.', default="Demisto Marketplace")
+    parser.add_argument('--index_path', help='The index.zip file path, generated on the cloud\n'
+                                             ' In case only_check_index_file is set, specify path to index.json',
+                        required=True)
+    parser.add_argument('--master_history', help='Path to a file that contains the master history commit hashes',
+                        required=True)
+    parser.add_argument('-s', '--secret', help='Path to secret conf file')
+
+    options = parser.parse_args()
+
+    return options
+
+
+def unzip_index_and_return_index_file(index_zip_path):
+    """Unzip index.zip and return the extracted index.json path
+
+    Args:
+        index_zip_path: The path to the index.zip file.
+
+    Returns: The extracted index.json path
+    """
+    logging.info('Unzipping')
+    with zipfile.ZipFile(index_zip_path, 'r') as zip_obj:
+        extracted_path = zip_obj.extract(member=f"index/{INDEX_FILE_PATH}", path="./extracted-index")
+
+    logging.info(f'extracted path is now: {extracted_path}')
+
+    return f"./{extracted_path}"
+
+
+def check_and_return_index_data(index_file_path):
+    """Check index.json file inside the index.zip archive in the cloud.
+
+    Validate no missing ids are found and that all packs have a positive price.
+
+    Args:
+        index_file_path: The path to the index.json.
+
+    Returns: Dict with the index data.
+
+    """
+    with open(index_file_path, 'r') as index_file:
+        index_data = json.load(index_file)
+
+    logging.info("Found index data in index file. Checking...")
+    logging.debug(f"Index data is:\n {index_data}")
+
+    assert len(index_data["packs"]) != 0, "Found 0 packs in index file."
+    for pack in index_data["packs"]:
+        assert pack["id"] != "", "There is a missing pack id."
```

Comment:
We should not use assert, let's gather all the errors that come up to a list and print them in a report at the end of the check.

Reply:
Done.